pkg/encryption/cipher_test.go

```diff
@@ -48,15 +32,199 @@ func TestEncodeAndDecodeAccessTokenB64(t *testing.T) {
 
 	secret, err := base64.URLEncoding.DecodeString(secretBase64)
 	assert.Equal(t, nil, err)
-	c, err := NewCipher([]byte(secret))
+	c, err := NewBase64Cipher(NewCFBCipher, []byte(secret))
 	assert.Equal(t, nil, err)
 
-	encoded, err := c.Encrypt(token)
+	encoded, err := c.Encrypt([]byte(token))
 	assert.Equal(t, nil, err)
 
 	decoded, err := c.Decrypt(encoded)
 	assert.Equal(t, nil, err)
 
-	assert.NotEqual(t, token, encoded)
-	assert.Equal(t, token, decoded)
+	assert.NotEqual(t, []byte(token), encoded)
+	assert.Equal(t, []byte(token), decoded)
+}
+
+func TestEncryptAndDecrypt(t *testing.T) {
+	// Test our 2 cipher types
+	ciphers := map[string]func([]byte) (Cipher, error){
+		"CFB": NewCFBCipher,
+		"GCM": NewGCMCipher,
+	}
+	for name, initCipher := range ciphers {
+		// Test all 3 valid AES sizes
+		for _, secretSize := range []int{16, 24, 32} {
+			subTestName := fmt.Sprintf("%s::%d", name, secretSize)
+			t.Run(subTestName, func(t *testing.T) {
```

Comment:
Nice, thanks for that! You can also change this a bit, I'm not sure if it's better or not though, so not blocking. WDYT?
Eg:

```
for name, initCipher := range ciphers {
  t.Run(name, func(t *testing.T) {
    // Test all 3 valid AES sizes
    for _, secretSize := range []int{16, 24, 32} {
      subTestName := fmt.Sprintf("%d", secretSize)
      t.Run(subTestName, func(t *testing.T) {
```
Each subtest appends the name so it could end up being `RUN TestEncryptAndDecrypt/CFB/16/10` for example depending on how deep we nest these